File: fixtures/scale/ruby/large/lib/check_jenkinshash.rb
require_relative 'jenkinshash_digest'
require_relative 'jenkinshash_digest_pair'
require_relative 'corpus_hash'

def check_jenkinshash
  corpus_hash.each do |text|
    first = jenkinshash_digest(text)
    second = jenkinshash_digest(text)
    return false if first != second || first.length != 8
  end
  true
end

File: fixtures/scale/python/large/cipherlab/checks/check_jenkinshash.py
from cipherlab.ciphers.jenkinshash import jenkinshash_digest
from cipherlab.specs.jenkinshash_spec import jenkinshash_spec
from cipherlab.corpus.corpus_hash import corpus_hash


def check_jenkinshash():
    spec = jenkinshash_spec()
    for text in corpus_hash():
        first = jenkinshash_digest(text)
        second = jenkinshash_digest(text)
        if first != second or len(first) != 8:
            return False
    return spec["category"] == "hash"
